src/Symfony/Component/EventDispatcher/Debug/TraceableEventDispatcher.php

```diff
@@ -150,8 +150,8 @@ public function dispatch($event/*, string $eventName = null*/)
             $event = $eventName ?? new Event();
             $eventName = $swap;
 
-            if (!$event instanceof Event) {
-                throw new \TypeError(sprintf('Argument 1 passed to "%s::dispatch()" must be an instance of %s, %s given.', EventDispatcherInterface::class, Event::class, \is_object($event) ? \get_class($event) : \gettype($event)));
+            if (!$event instanceof Event && !$event instanceof ContractsEvent) {
```

Comment:
This looks wing to me. The legacy signature did not support passing objects as events that do not extend the `Event` class from the component.